wwwapp/models.py

```diff
@@ -59,6 +59,47 @@ class WorkshopUserProfile(models.Model):
     def __str__(self):
         return '%s: %s, %s' % (self.year, self.user_profile, self.status)
 
+class PESELField(models.CharField):
+    """PESEL field, with checksum verification."""
+
+    def validate(self, pesel: str, user_info: 'UserInfo') -> None:
```

Comment:
Are you sure this signature is right? Seems to be only `self, value`
https://github.com/django/django/blob/master/django/forms/fields.py#L125

Reply:
I'm not sure how it works, but locally, with `django==1.11.22` I got error otherwise:
```
TypeError: validate() missing 1 required positional argument: 'model_instance'
```

That's probably because the appropriate file in this case is https://github.com/django/django/blob/master/django/db/models/fields/__init__.py

Reply:
Oh right, I was looking at the wrong file D: